src/Logger.js

```diff
@@ -0,0 +1,102 @@
+const pino = require("pino");
+
+/**
+ * Wrapper class around pino package to add some extra functionality.
+ */
+module.exports = class Logger {
+    /**
+     * @param {string} loglevel  Accepted values: fatal | error | warn | info | debug | trace
+     * @param {object} context   Common props to add to all logged messages.
+     * @param {object} logger    Pino instance to use, if null create a new one (default).
+     */
+    constructor(loglevel = "info", context = {}, logger = null) {
+        this._logger = logger || pino({
+            messageKey: "message",
+            base:       null,
+            timestamp:  true,
+            level:      loglevel
+        });
+        this.setLogLevel(loglevel);
+        this.setContext(context);
+    }
+
+    /**
+     * Sets the pino-compatible log level to be used from this point forward.
+     *
+     * @param {string} loglevel  Accepted values: fatal | error | warn | info | debug | trace
+     */
+    setLogLevel(loglevel) {
+        this._logger.level = loglevel;
+    }
+
+    /**
+     * Sets common props that will be added to all logs from this point forward.
+     * Useful for things like request_id, type/namespace/service, etc.
+     *
+     * @param {object} context   Common props to add to all logged messages.
+     */
+    setContext(context) {
+        this._context = context;
+    }
+
+    /**
+     * Adds/overwrites common props that will be added to all logs from this point forward.
+     * Useful for things like request_id, type/namespace/service, etc.
+     *
+     * @param {object} context   Common props to add to all logged messages.
+     */
+    addContext(context) {
+        this._context = {...this._context, ...context};
+    }
+
+    /**
+     * Creates a child logger from the current one, inherting
+     *
+     * @param {object} context   Common props to add to all logged messages.
+     */
+    child(options) {
+        return new Logger(this._logger.level, this._context, this._logger.child(options));
```

Comment:
Wouldn't this be the exact equivalent of

`return new Logger(this._logger.level, { ...this._context, ...options});`

but without the need of passing down another instance of pino?

Reply:
in theory yes, in practice we use `pino.child()` in the adstitcher and wasn't sure if there was any other side-effects of not doing it like this. The only difference I see is that the `options` passed would be "built-into" the logger and would not be able to later be changed via `setContext()`, since they'd be part of the pino instance. Since we do this in the adstitcher and wanted to keep the exact behavior but in theory, yes your alternative should do the same.

Reply:
I'd say it's a matter of consistency. I'm fine with both using a local context or `pino.child` (as I was saying it's an optimization) but I'd be consistent and use always either one of the two

Reply:
@roccozanni will change it for the sake of consistency, you're right